### PR TITLE
fix: disable access to external resources and functions for securization

### PR DESCRIPTION
**Pull request by marcambier**
fix: disable access to external resources and functions for securization

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `1.6.4-fix-securizeWsdlPolicy-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-xslt/1.6.4-fix-securizeWsdlPolicy-SNAPSHOT/gravitee-policy-xslt-1.6.4-fix-securizeWsdlPolicy-SNAPSHOT.zip)

### CODE REVIEW

File: src/main/java/io/gravitee/policy/xslt/transformer/saxon/SaxonTransformerFactory.java [hunk 1]
```diff
@@ -16,28 +16,32 @@
...
  * @author GraviteeSource Team
  */
-public class SaxonTransformerFactory extends TransformerFactoryImpl {
 
-    @Override
-    public Transformer newTransformer(Source arg0) throws TransformerConfigurationException {
-        this.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
-        Transformer t = super.newTransformer(arg0);
-        return t;
+public class SaxonTransformerFactory extends net.sf.saxon.TransformerFactoryImpl {
```
Comment (Okhelifi) [inline src/main/java/io/gravitee/policy/xslt/transformer/saxon/SaxonTransformerFactory.java:29]:
```suggestion
public class SaxonTransformerFactory extends TransformerFactoryImpl {
```

File: src/main/java/io/gravitee/policy/xslt/transformer/saxon/SaxonTransformerFactory.java [hunk 2]
```diff
@@ -16,28 +16,32 @@
...
-        this.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
-        Transformer t = super.newTransformer();
-        return t;
+    private void enableSecureProcessing() {
+        try {
+            this.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            this.setAttribute(FeatureKeys.ALLOWED_PROTOCOLS, "");
+            this.setAttribute(FeatureKeys.ALLOW_EXTERNAL_FUNCTIONS, false);
+        } catch (TransformerConfigurationException e) {
+            LOGGER.error("Failed to set SaxonTransformer feature", e);
```
Comment (Okhelifi) [inline src/main/java/io/gravitee/policy/xslt/transformer/saxon/SaxonTransformerFactory.java:44]: Do you think we should throw an exception here ?
On the one hand if we catch the exception and throw nothing it means that we can have a vulnerability. On the other hand we should never catch any exception here I think...

**Review by Okhelifi — APPROVED** [review]

**Comment by Okhelifi** [comment]
Good Job 👏🏼

File: src/main/java/io/gravitee/policy/xslt/transformer/saxon/SaxonTransformerFactory.java (hunk 1 above)
Comment (marcambier) [reply src/main/java/io/gravitee/policy/xslt/transformer/saxon/SaxonTransformerFactory.java]: :heavy_check_mark:

File: src/main/java/io/gravitee/policy/xslt/transformer/saxon/SaxonTransformerFactory.java (hunk 2 above)
Comment (marcambier) [reply src/main/java/io/gravitee/policy/xslt/transformer/saxon/SaxonTransformerFactory.java]: yep, it's better with an exception to ensure it's securized. I changed it :+1:

**Pull request merged**